web/src/main/java/org/openmrs/web/Listener.java

```diff
@@ -250,6 +250,32 @@ public void contextInitialized(ServletContextEvent event) {
 				servletContext.setAttribute(WebApplicationContext.ROOT_WEB_APPLICATION_CONTEXT_ATTRIBUTE, context);
 				
 				WebDaemon.startOpenmrs(event.getServletContext());
+
+                // ✅ Print the access URL
+				final String contextPath = servletContext.getContextPath();
+
+				String portTmp = System.getProperty("jetty.port");
+				if (portTmp == null || portTmp.isEmpty()) {
+					portTmp = System.getProperty("server.port");
+				}
+				if (portTmp == null || portTmp.isEmpty()) {
+					portTmp = "8080"; // fallback
+				}
+				final String port = portTmp;
+				
+				final String host = "http://localhost";
+				
+				new Thread(() -> {
+					try {
+						Thread.sleep(3000); // Let Jetty finish logging
+						System.out.println("✅ Server running at: " + host + ":" + port + contextPath);
```

Comment:
https://openmrs.atlassian.net/wiki/spaces/docs/pages/25477044/Java+Conventions#Do-not-print-out